docs/guides/directing-traffic-to-your-app.md

```diff
@@ -0,0 +1,375 @@
+
+By default, Gateways only allow routes (such as HTTPRoutes) to attach if they are in the same namespace as the Gateway.
+If you want to change this behavior, you can set
+the [`allowedRoutes`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.AllowedRoutes)
+field.
+
+Now, let's create the HTTPRoute by copying and pasting the following into your terminal:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: coffee
+spec:
+  parentRefs:
+  - name: cafe
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /
+    backendRefs:
+    - name: coffee
+      port: 80
+EOF
+```
+
+To attach the `coffee` HTTPRoute to the `cafe` Gateway, we specify the Gateway name in the `parentRefs` field. The
+attachment will succeed if the hostnames and protocol in the HTTPRoute are allowed by at least one of the
+Gateway's listeners.
+
+The `hostnames` field allows you to list the hostnames that the HTTPRoute matches. In this case, incoming requests
+handled by the `http` listener with the HTTP host header `cafe.example.com` will match this HTTPRoute and will be routed
+according to the rules in the spec.
+
+The `rules` field defines routing rules for the HTTPRoute. A rule is selected if the request satisfies one of the
+rule's `matches`. To forward traffic for all paths to the coffee Service we specify a match with the PathPrefix `/` and
+target the coffee Service using the `backendRef` field.
+
+## Test the Configuration
+
+To test the configuration, we will send a request to the public IP and port of NGINX Kubernetes Gateway that you saved
+in the [prerequisites](#prerequisites) section and verify that the response comes from one of the coffee Pods.
+
+> **Note**
+> In a production environment, the coffee application would be accessible through the hostname `cafe.example.com`.
+> However, in this guide, we will be accessing the application through the public IP and port of the NGINX Kubernetes
+> Gateway Service and with curl's `--resolve` option.
+
+
+First, let's send a request to the path `/`:
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/
+```
+
+We should get a response from one of the coffee Pods:
+
+```text
+Server address: 10.12.0.18:8080
+Server name: coffee-7dd75bc79b-cqvb7
+```
+
+Since the `cafe` HTTPRoute routes all traffic on any path to the coffee application, the following requests should also
+be handled by the coffee Pods:
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/some-path
+```
+
+```text
+Server address: 10.12.0.18:8080
+Server name: coffee-7dd75bc79b-cqvb7
+```
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/some/path
+```
+
+```text
+Server address: 10.12.0.19:8080
+Server name: coffee-7dd75bc79b-dett3
+```
```

Comment:
would it make sense to use a different hostname like `pub.example.com` and to show that NKG will return 404 in this case?

Reply:
Added. Not sure about the wording so let me know what you think